Prog/NCHousing_needs_units_targets.sas

```diff
@@ -0,0 +1,927 @@
+by MID;
+run;
+
+data Housing_needs_baseline_&year._2;
+merge Housing_needs_baseline_&year._1(in=a) NC17_limits;
+if a;
+by MID;
+run;
+
+data Housing_needs_baseline_&year.;
+set Housing_needs_baseline_&year._2;
+	  if hhincome ~=.n or hhincome ~=9999999 then do; 
+		 %dollar_convert( hhincome, hhincome_a, &year., 2017, series=CUUR0000SA0 )
+	   end; 
+
+	*create HUD_inc - uses 2017 limits but has categories for 120-200% and 200%+ AMI; 
+
+  %Hud_inc_NCState( hhinc=hhincome_a, hhsize=numprec )  
+
+run; 
+
+data Housing_needs_baseline_&year._3;
+  set Housing_needs_baseline_&year.;
+
+	 *adjust housing costs for inflation; 
+
+	  %dollar_convert( rentgrs, rentgrs_a, &year., 2017, series=CUUR0000SA0L2 )
+	  %dollar_convert( owncost, owncost_a, &year., 2017, series=CUUR0000SA0L2 )
+	  %dollar_convert( valueh, valueh_a, &year., 2017, series=CUUR0000SA0L2 )
+
+  	** Cost-burden flag & create cost ratio **;
+	    if ownershpd in (21, 22)  then do;
+
+			if hhincome_a > 0 then Costratio= (rentgrs_a*12)/hhincome_a;
+			  else if hhincome_a = 0 and rentgrs_a > 0 then costratio=1;
+			  else if hhincome_a =0 and rentgrs_a = 0 then costratio=0; 
+			  else if hhincome_a < 0 and rentgrs_a >= 0 then costratio=1; 
+			  			  
+		end;
+
+	    else if ownershpd in ( 12,13 ) then do;
+			if hhincome_a > 0 then Costratio= (owncost_a*12)/hhincome_a;
+			  else if hhincome_a = 0 and owncost_a > 0 then costratio=1;
+			  else if hhincome_a =0 and owncost_a = 0 then costratio=0; 
+			  else if hhincome_a < 0 and owncost_a >= 0 then costratio=1; 
+		end;
+	    
+			if Costratio >= 0.3 then costburden=1;
+		    else if HHIncome_a~=. then costburden=0;
+			if costratio >= 0.5 then severeburden=1;
+			else if HHIncome_a~=. then severeburden=0; 
+
+		tothh = 1;
+
+    
+    ****** Rental units ******;
+    
+   if ownershpd in (21, 22) then do;
+        
+    Tenure = 1;
+
+	 *create maximum desired or affordable rent based on HUD_Inc categories*; 
+    /* need to discuss for NC, use hudinc for now*/
+	  if hud_inc in(1 2 3) then max_rent=HHINCOME_a/12*.3; *under 80% of AMI then pay 30% threshold; 
+	  if hud_inc =4 then max_rent=HHINCOME_a/12*.25; *avg for all HH hud_inc=4; 
+	  if costratio <=.18 and hud_inc = 5 then max_rent=HHINCOME_a/12*.18; *avg for all HH hud_inc=5; 	
+		else if hud_inc = 5 then max_rent=HHINCOME_a/12*costratio; *allow 120-200% above average to spend more; 
+	  if costratio <=.12 and hud_inc = 6 then max_rent=HHINCOME_a/12*.12; *avg for all HH hud_inc=6; 
+	  	else if hud_inc=6 then max_rent=HHINCOME_a/12*costratio; *allow 200%+ above average to spend more; 
+     
+	 *create flag for household could "afford" to pay more; 
+		couldpaymore=.;
+
+		if max_rent ~= . then do; 
+			if max_rent > rentgrs_a*1.1 then couldpaymore=1; 
+			else if max_rent <= rentgrs_a*1.1 then couldpaymore=0; 
+		end; 
+
+	
+    	*rent cost categories that make more sense for rents - no longer used in targets;
+		/*need to discuss for NC*/
+			rentlevel=.;
```

Comment:
these don't look right to me. i think you want to start the categories much lower  - 30% of income for an ELI renter (statewide) is $443. the categories don't have to precisely align with AMI but the bottom 3 categories should be close to 0-30% of AMI, 30-50% and 50-80%.

Reply:
how did you came up with the $443 number?

Reply:
hmm not sure. use the family of 4 categories to come up with some ranges. don't have to match precisely.